src/main/java/org/cbioportal/web/columnar/ClinicalDataBinner.java

```diff
@@ -0,0 +1,121 @@
+package org.cbioportal.web.columnar;
+
+import org.cbioportal.model.*;
+import org.cbioportal.service.StudyViewColumnarService;
+import org.cbioportal.web.columnar.util.NewClinicalDataBinUtil;
+import org.cbioportal.web.parameter.*;
+import org.cbioportal.web.util.DataBinner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class ClinicalDataBinner {
+    private final StudyViewColumnarService studyViewColumnarService;
+    private final DataBinner dataBinner;
+    
+    @Autowired
+    public ClinicalDataBinner(
+        StudyViewColumnarService studyViewColumnarService,
+        DataBinner dataBinner
+    ) {
+        this.studyViewColumnarService = studyViewColumnarService;
+        this.dataBinner = dataBinner;
+    }
+
+    public List<ClinicalDataBin> fetchClinicalDataBinCounts(
+        DataBinMethod dataBinMethod,
+        ClinicalDataBinCountFilter dataBinCountFilter,
+        boolean shouldRemoveSelfFromFilter
+    ) {
+        List<ClinicalDataBinFilter> attributes = dataBinCountFilter.getAttributes();
+        StudyViewFilter studyViewFilter = dataBinCountFilter.getStudyViewFilter();
+
+        if (shouldRemoveSelfFromFilter) {
+            studyViewFilter = NewClinicalDataBinUtil.removeSelfFromFilter(dataBinCountFilter);
+        }
+
+        List<String> attributeIds = attributes.stream().map(ClinicalDataBinFilter::getAttributeId).collect(Collectors.toList());
+
+        // a new StudyView filter to partially filter by study and sample ids only
```

Comment:
@onursumer can you put some comments here explaining why we do this?  I know this is the same logic as legacy but it's not clear to me why we first do this partial filtering.